### PR TITLE
Add script to convert brat standoff to dygiepp format

### PR DESCRIPTION
**Pull request by serenalotreck**
First attempt at a script to convert the brat standoff format (.ann files) to the required input for dygiepp. I've only done some bare-bones testing to get the code up and running in `dygie/tests/data/annotated_doc_test.py`, and haven't tested many edge cases, so if anyone wants to add any tests, please feel free! I'd appreciate any feedback on the structure/approach of the code, and I've left some notes in the docstrings for functions that I'd especially like feedback on.

### CODE REVIEW

**Comment by serenalotreck** [comment]
Whoops, I meant to revert commit `6d5337d` before opening this PR, apologies! Feel free to revert it, it doesn't have to do with the brat conversion.

**Comment by dwadden** [comment]
Thanks! I'll take a look over the weekend (but probably not before, unfortunately). 

Not sure if you've done this already, but if you haven't, it would be great to have a test case to confirm that things are working as you expect.   This could be, say, a document consisting of couple sentences, together with the `.ann` file for that document and the expected output of running your code on it. The input and output files can go in a new subfolder in the [tests/fixtures](https://github.com/dwadden/dygiepp/tree/master/dygie/tests/fixtures) directory, and the test code can go in [tests/data](https://github.com/dwadden/dygiepp/tree/master/dygie/tests/data). If you've already got this, even better.

Thanks again; will respond once I've taken a look.

**Comment by serenalotreck** [comment]
No rush at all, thanks for getting to it so quickly!

I do have a test case (or set thereof) in `/dygie/tests/data/annotated_doc.test.py`, I just used the minimal example from `data.md` that you use to illustrate the formatting for the different annotation types (I have it make the files it needs in `setUp()`). It passed all the tests, but there are plenty of edge cases I didn't include -- happy to add some if you'd like!

**Comment by dwadden** [comment]
Sorry I haven't gotten to this yet! I'm wrapping up an internship project and may be busy till that ends in two weeks, but I will get back to this. 

Is this PR blocking you from getting your own work done? If so, let me know which part is blocking and I can at least look at that.

**Comment by serenalotreck** [comment]
No worries at all! I can just use my own fork if I need it in the meantime, but I'm currently stuck on an upstream step so I haven't had to use it yet. Looking forward to having your eyes on it!

File: scripts/new-dataset/annotated_doc.py [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+        tokens in the list in order to convert char indices to token indices.
+        However, because there usually aren't spaces before punctuation like 
+        periods and commas, but there can be before hyphens, I wasn't sure how 
+        to make sure that I didn't make an asusmption about where to insert
+        spaces when doing the cumulative sum, because even one mismatch (e.g.
+        if I assumed there shouldn't be a space between a hyphen and the 
+        last token, but then a text uses a hypen as an en dash) would cause
+        all of the rest of the entities in the doc to be thrown away. 
+
+        So the approach I settled on was to order the entities by their
```
Comment (dwadden) [inline scripts/new-dataset/annotated_doc.py:212]: If what you'd done works, seems fine to leave it. An alternative approach would be to rely on Spacy to do the indexing for you. When you process a document with Spacy, it computes start and end indices for each token, with respect to both the whole document and the sentence that the token is part of. So, if you know the character start and end indices of the entity you're looking for, you could just search for the Spacy tokens that match.

```
nlp = spacy.load("en_core_web_sm")
doc = nlp("Here's one sentence. And another sentence.")
# Skip to the last sentence.
for sent in doc.sents: pass
# Print a single word in the second sentence
print(sent[1])
# Print start character index of word in document
print(sent[1].idx)
# Print start token index of word in document
print(sent[1].i)
# Print start token index of word in sentence
print(sent[1].sent_start)
```

File: dygie/tests/data/annotated_doc_test.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    def test_format_corefs_dygiepp(self):
+
+        self.equivrel1.set_arg_objects(self.ents)
+        self.equivrel2.set_arg_objects(self.ents)
+        corefs = ad.EquivRel.format_corefs_dygiepp([self.equivrel1, self.equivrel2])
+
+        self.assertEqual(corefs, self.corefs)
+
+
+class TestAnnotatedDoc:
```
Comment (dwadden) [inline dygie/tests/data/annotated_doc_test.py:190]: Should subclass `unittest.TestCase`; otherwise it doesn't run when invoked using `python -m unittest annotated_doc_test.py`.

Comment (dwadden) [reply dygie/tests/data/annotated_doc_test.py:190]: This test broke when run, with error 

```
======================================================================
ERROR: test_format_dygiepp (annotated_doc_test.TestAnnotatedDoc)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/dwadden/proj/dygiepp/dygie/tests/data/annotated_doc_test.py", line 250, in test_format_dygiepp
    annotated_doc = ad.AnnotatedDoc.parse_ann(self.txt, self.ann, self.nlp, self.dataset)
TypeError: parse_ann() missing 1 required positional argument: 'coref'
```

**Review by dwadden — CHANGES_REQUESTED** [review]
OK, I took a look. Generally looks good to me. Comments inline. A couple additional, general things:
- Strip whitespace, run a linter and a code formatter (take your pick).
- Add a section to `doc/data.md` explaining how to use this.

File: scripts/new-dataset/annotated_doc.py (hunk 1 above)
Comment (serenalotreck) [reply scripts/new-dataset/annotated_doc.py]: Thanks for the feedback! I changed it to use spacy, as that felt safer/more robust.
